cmd/collector/app/processor/span.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import (
+	"github.com/jaegertracing/jaeger/model"
+)
+
+// SpansOptions additional options passed to processor along with the spans.
+type SpansOptions struct {
+	SpanFormat       SpanFormat
+	InboundTransport InboundTransport
+}
+
+// Span handles model spans
+type Span interface {
```

Comment:
This one was just moved from a "generic" source file from `cmd/collector/app`. It has been moved here to avoid a cyclic dependency. We could perhaps do some refactoring around the processor part as well in a follow-up PR.

Reply:
I think that keeping the old name of `SpanProcessor` (and having a stuttering type name) is fine and more readable in this instance.

Reply:
`processor.SpanProcessor#ProcessSpans()` seems quite repetitive, but I agree with reverting this change, especially because it's not the focus of this PR.